### PR TITLE
xbutil2: Complete abstraction library query functionality

### PR DESCRIPTION
**Pull request by rozumx**
Work Done
    ---------
    + Implemented multilay device_core <|-- device_pci <|-- device_linux or device_windows abstraction layers
    + Abstracted all query calls to enumeration values
    + Abstracted all sysdev or IOCTL calls to the lower level layers (e.g. device_linux and device_windows)
    + Added helper methods to collect query data and format it into a property table
    + Added stub body file non linux/window libraries that depends on core_common
    + On windows, the target executable is now xbutil
    + Refactored how sub-commands register themself to the xbutil main().
      - It is now cmake driven
    + Updated CMake to only build program, query, clock, reset, and version sub-commands
    + Sub-command help is now passed along with the other options

### CODE REVIEW

**Comment by rozumx** [comment]
please retest

**Comment by rozumx** [comment]
retest this please

File: src/runtime_src/core/tools/xbutil2/SubCmdFan.cpp [hunk 1]
```diff
@@ -27,23 +27,33 @@ namespace po = boost::program_options;
 // System - Include Files
 #include <iostream>
 
+// ======= R E G I S T E R   T H E   S U B C O M M A N D ======================
+#include "SubCmd.h"
+static const unsigned int registerResult = 
+                    register_subcommand("fan",
```
Comment (maxzhen) [inline src/runtime_src/core/tools/xbutil2/SubCmdFan.cpp]: just curious, we are able to set fan speed from command line?

File: src/runtime_src/core/tools/xbutil2/SubCmdP2P.cpp [hunk 2]
```diff
@@ -27,14 +27,22 @@ namespace po = boost::program_options;
 // System - Include Files
 #include <iostream>
 
+// ======= R E G I S T E R   T H E   S U B C O M M A N D ======================
+#include "SubCmd.h"
+static const unsigned int registerResult = 
+                    register_subcommand("p3p",
```
Comment (maxzhen) [inline src/runtime_src/core/tools/xbutil2/SubCmdP2P.cpp]: you mean "p2p"?

File: src/runtime_src/core/tools/xbutil2/SubCmdRun.cpp [hunk 3]
```diff
@@ -27,24 +27,34 @@ namespace po = boost::program_options;
 // System - Include Files
 #include <iostream>
 
+// ======= R E G I S T E R   T H E   S U B C O M M A N D ======================
+#include "SubCmd.h"
+static const unsigned int registerResult = 
+                    register_subcommand("run",
```
Comment (maxzhen) [inline src/runtime_src/core/tools/xbutil2/SubCmdRun.cpp]: not sure what "run" is for?

File: src/runtime_src/core/tools/xbutil2/SubCmdScan.cpp [hunk 4]
```diff
@@ -27,22 +27,35 @@ namespace po = boost::program_options;
 // System - Include Files
 #include <iostream>
 
+// ======= R E G I S T E R   T H E   S U B C O M M A N D ======================
+#include "SubCmd.h"
+static const unsigned int registerResult = 
+                    register_subcommand("scan",
```
Comment (maxzhen) [inline src/runtime_src/core/tools/xbutil2/SubCmdScan.cpp]: in xbutil, list is the same as scan and i actually removed "list" from help message and only left scan. maybe we should do the same for xbutil2 as well?

**Comment by abalasa** [comment]
retest this please

File: src/runtime_src/core/tools/xbutil2/SubCmdP2P.cpp (hunk 2 above)
Comment (rozumx) [reply src/runtime_src/core/tools/xbutil2/SubCmdP2P.cpp]: Corrected subcommand name to be: p2p

File: src/runtime_src/core/tools/xbutil2/SubCmdScan.cpp (hunk 4 above)
Comment (rozumx) [reply src/runtime_src/core/tools/xbutil2/SubCmdScan.cpp]: Great to know!  Will make those changes as well.

File: src/runtime_src/core/tools/xbutil2/SubCmdFan.cpp (hunk 1 above)
Comment (rozumx) [reply src/runtime_src/core/tools/xbutil2/SubCmdFan.cpp]: The "fan" and "run" subcommands were added prior to examining to see if they are actually implemented.  I will remove them from consideration.

File: src/runtime_src/core/tools/xbutil2/SubCmdClock.cpp [hunk 5]
```diff
@@ -74,7 +85,7 @@ int subCmdClock(const std::vector<std::string> &_options, bool _help)
   }
 
   // Check to see if help was requested or no command was found
-  if (_help == true)  {
+  if (help == true)  {
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdClock.cpp]: Is the bool_switch working? I tried the same implementation in xbmgmt, but it doesn't seem to work. It works under 2 scenarios:
1. change help=true in the beginning and this condition to check if help=false
2. or change --help to -h

**Pull request merged**